<a href="https://colab.research.google.com/github/bentemarij/DSRP-Small-Body-Close-Approaches-BMdZ/blob/main/Bente_de_Zwart_DSRP_Colab_Small_Body_Close_Approaches.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 50)

In [ ]:
df = pd.read_csv('https://raw.githubusercontent.com/frontiertechinstitute/datasets/refs/heads/main/SB%20Close%20Approaches/earth_close_approaches_50yr.csv')
df.info()
print(df.describe())
df

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 186255 entries, 0 to 186254
Data columns (total 8 columns):
 #   Column    Non-Null Count   Dtype  
---  ------    --------------   -----  
 0   des       186255 non-null  object 
 1   cd        186255 non-null  object 
 2   dist      186255 non-null  float64
 3   dist_min  186255 non-null  float64
 4   dist_max  186255 non-null  float64
 5   v_rel     186255 non-null  float64
 6   h         186065 non-null  float64
 7   diameter  5375 non-null    float64
dtypes: float64(6), object(2)
memory usage: 11.4+ MB
                dist       dist_min       dist_max          v_rel  \
count  186255.000000  186255.000000  186255.000000  186255.000000   
mean        0.237560       0.223031       0.256043      12.905789   
std         0.147360       0.144774       0.156881       6.684920   
min         0.000014       0.000000       0.000025       0.056485   
25%         0.109646       0.095617       0.122456       7.764031   
50%         0.228977   

,des,cd,dist,dist_min,dist_max,v_rel,h,diameter
0,2015 RT1,1976-Jan-01 04:35,0.111611,0.109327,0.113897,11.086209,25.410,NaN
1,2025 JH2,1976-Jan-01 06:22,0.171429,0.171313,0.171548,5.842427,27.050,NaN
2,2001 KU66,1976-Jan-01 14:29,0.440401,0.426042,0.466510,10.401878,24.100,NaN
3,2015 TG24,1976-Jan-01 15:04,0.451030,0.425241,0.477393,12.519296,26.680,NaN
4,2002 TX59,1976-Jan-02 01:26,0.459706,0.459668,0.459744,6.299898,23.900,NaN
...,...,...,...,...,...,...,...,...
186250,2026 LD1,2026-Jun-12 19:50,0.130207,0.128030,0.132384,10.184379,23.314,NaN
186251,2026 LK1,2026-Jun-12 21:43,0.005988,0.005983,0.005993,5.837560,27.872,NaN
186252,388945,2026-Jun-12 22:12,0.112264,0.112264,0.112264,7.484532,20.440,NaN
186253,2018 NC15,2026-Jun-12 22:50,0.131353,0.131353,0.131354,10.225792,21.810,NaN


**Possible Issues**

May be some issues with the 'des' column, there are 2 formats for the name of the small body object. Can be either their provisional or permanent name.
* Ex: line 0 - 2015 RT1 (provisional), line 15 - 523808 (permanent)

May be some issues with the 'diameter' column, as many of the values are NaN.
* 5375 non-null out of 186255 lines (dtype: float64)

In [ ]:
df['des'].value_counts()

,count
des,
469219,101
2025 PN7,83
2017 FZ2,81
2023 FW13,81
2021 EY1,78
...,...
2026 LD1,1
2026 LT,1
2026 KT2,1


Each of the rows does not represent a small body, it respresents a close approach. The most approaches made by a single body is 101 by 469219. After quick searches it seems like the majority of the top of this list are Earth's qausi moons.

In [ ]:
df469219 = df[df['des'] == '469219']
df469219.tail(15)

,des,cd,dist,dist_min,dist_max,v_rel,h,diameter
139565,469219,2019-Apr-19 16:17,0.150700,0.150699,0.150701,4.117841,24.31,NaN
142279,469219,2019-Oct-08 04:43,0.151210,0.151210,0.151211,4.331279,24.31,NaN
145682,469219,2020-Apr-13 21:23,0.155974,0.155973,0.155975,4.046718,24.31,NaN
148897,469219,2020-Oct-12 08:33,0.146759,0.146758,0.146760,4.423755,24.31,NaN
152309,469219,2021-Apr-08 12:39,0.160722,0.160721,0.160723,3.950523,24.31,NaN
155836,469219,2021-Oct-17 11:30,0.141673,0.141672,0.141674,4.508019,24.31,NaN
159120,469219,2022-Apr-02 18:16,0.164750,0.164749,0.164751,3.821477,24.31,NaN
162786,469219,2022-Oct-22 04:42,0.136348,0.136348,0.136349,4.549048,24.31,NaN
165776,469219,2023-Mar-27 18:56,0.167866,0.167865,0.167867,3.658830,24.31,NaN
169772,469219,2023-Oct-26 13:35,0.130763,0.130762,0.130763,4.592505,24.31,NaN


**Accurate Columns**

After comparing the df469219 with information about 469219 and also finding the official NASA data base, I can now accurately determine each of the columns:
* `des` - primary designation of the asteroid or comet (e.g., 443, 2000 SG344)
* `cd` - time of close-approach (formatted calendar date/time, TDB)
* `dist` - nominal approach distance (au)
* `dist_min` - minimum (3-sigma) approach distance (au)
* `dist_max` - maximum (3-sigma) approach distance (au)
* `v_rel` - velocity relative to the approach body at close approach (km/s)
* `h` - absolute magnitude H (mag)
* `diameter` - diameter of the body (km)
  * `NaN` if not known

---

Sources/links used:

* https://www.spacereference.org/asteroid/469219-kamooalewa-2016-ho3
* https://ssd-api.jpl.nasa.gov/doc/cad.html


